### ISSUE TITLE
SyntaxError: missing ) after argument list

### ISSUE THREAD

**Issue by shaunluttin** [Izhaki/Typescript-Jasmine-Istanbul-Boilerplate#1]
# Steps to Reproduce

```
git clone git@github.com:Izhaki/Typescript-Jasmine-Istanbul-Boilerplate.git
cd Typescript-Jasmine-Istanbul-Boilerplate
npm install -y
npm run test:coverage
```
# Error

```
> typescript-jasmine-istanbul-boilerplate@1.0.0 test:coverage C:\temp\Typescript-Jasmine-Istanbul-Boilerplate
> ts-node node_modules/istanbul/lib/cli.js cover -e .ts  -x "*.d.ts" -x "*.spec.ts" node_modules/.bin/jasmine -- JASMINE_CONFIG_PATH=jasmine.json

SyntaxError: missing ) after argument list
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:373:25)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
    at Function.Module.runMain (module.js:441:10)
    at runFn (C:\temp\Typescript-Jasmine-Istanbul-Boilerplate\node_modules\istanbul\lib\command\cover.js:126:16)
    at C:\temp\Typescript-Jasmine-Istanbul-Boilerplate\node_modules\istanbul\lib\command\cover.js:146:9
    at C:\temp\Typescript-Jasmine-Istanbul-Boilerplate\node_modules\istanbul\node_modules\istanbul-api\lib\run-cover.js:187:16
    at C:\temp\Typescript-Jasmine-Istanbul-Boilerplate\node_modules\istanbul\node_modules\istanbul-api\lib\file-matcher.js:69:16
No coverage information was collected, exit without writing coverage information

npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "test:coverage"
npm ERR! node v4.4.7
npm ERR! npm  v2.15.8
npm ERR! code ELIFECYCLE
npm ERR! typescript-jasmine-istanbul-boilerplate@1.0.0 test:coverage: `ts-node node_modules/istanbul/lib/cli.js cover -e .ts  -x "*.d.ts" -x "*.spec.ts" node_modules/.bin/jasmine -- JASMINE_CONFIG_PATH=jasmine.json`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the typescript-jasmine-istanbul-boilerplate@1.0.0 test:coverage script 'ts-node node_modules/istanbul/lib/cli.js cover -e .ts  -x "*.d.ts" -x "*.spec.ts" node_modules/.bin/jasmine -- JASMINE_CONFIG_PATH=jasmine.json'.
npm ERR! This is most likely a problem with the typescript-jasmine-istanbul-boilerplate package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     ts-node node_modules/istanbul/lib/cli.js cover -e .ts  -x "*.d.ts" -x "*.spec.ts" node_modules/.bin/jasmine -- JASMINE_CONFIG_PATH=jasmine.json
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs typescript-jasmine-istanbul-boilerplate
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!
npm ERR!     npm owner ls typescript-jasmine-istanbul-boilerplate
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\temp\Typescript-Jasmine-Istanbul-Boilerplate\npm-debug.log
```
# npm-debug.log

```
0 info it worked if it ends with ok
1 verbose cli [ 'C:\\Program Files\\nodejs\\node.exe',
1 verbose cli   'C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'run',
1 verbose cli   'test:coverage' ]
2 info using npm@2.15.8
3 info using node@v4.4.7
4 verbose run-script [ 'pretest:coverage', 'test:coverage', 'posttest:coverage' ]
5 info pretest:coverage typescript-jasmine-istanbul-boilerplate@1.0.0
6 info test:coverage typescript-jasmine-istanbul-boilerplate@1.0.0
7 verbose unsafe-perm in lifecycle true
8 info typescript-jasmine-istanbul-boilerplate@1.0.0 Failed to exec test:coverage script
9 verbose stack Error: typescript-jasmine-istanbul-boilerplate@1.0.0 test:coverage: `ts-node node_modules/istanbul/lib/cli.js cover -e .ts  -x "*.d.ts" -x "*.spec.ts" node_modules/.bin/jasmine -- JASMINE_CONFIG_PATH=jasmine.json`
9 verbose stack Exit status 1
9 verbose stack     at EventEmitter.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\lifecycle.js:217:16)
9 verbose stack     at emitTwo (events.js:87:13)
9 verbose stack     at EventEmitter.emit (events.js:172:7)
9 verbose stack     at ChildProcess.<anonymous> (C:\Program Files\nodejs\node_modules\npm\lib\utils\spawn.js:24:14)
9 verbose stack     at emitTwo (events.js:87:13)
9 verbose stack     at ChildProcess.emit (events.js:172:7)
9 verbose stack     at maybeClose (internal/child_process.js:827:16)
9 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:211:5)
10 verbose pkgid typescript-jasmine-istanbul-boilerplate@1.0.0
11 verbose cwd C:\temp\Typescript-Jasmine-Istanbul-Boilerplate
12 error Windows_NT 10.0.14393
13 error argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "test:coverage"
14 error node v4.4.7
15 error npm  v2.15.8
16 error code ELIFECYCLE
17 error typescript-jasmine-istanbul-boilerplate@1.0.0 test:coverage: `ts-node node_modules/istanbul/lib/cli.js cover -e .ts  -x "*.d.ts" -x "*.spec.ts" node_modules/.bin/jasmine -- JASMINE_CONFIG_PATH=jasmine.json`
17 error Exit status 1
18 error Failed at the typescript-jasmine-istanbul-boilerplate@1.0.0 test:coverage script 'ts-node node_modules/istanbul/lib/cli.js cover -e .ts  -x "*.d.ts" -x "*.spec.ts" node_modules/.bin/jasmine -- JASMINE_CONFIG_PATH=jasmine.json'.
18 error This is most likely a problem with the typescript-jasmine-istanbul-boilerplate package,
18 error not with npm itself.
18 error Tell the author that this fails on your system:
18 error     ts-node node_modules/istanbul/lib/cli.js cover -e .ts  -x "*.d.ts" -x "*.spec.ts" node_modules/.bin/jasmine -- JASMINE_CONFIG_PATH=jasmine.json
18 error You can get information on how to open an issue for this project with:
18 error     npm bugs typescript-jasmine-istanbul-boilerplate
18 error Or if that isn't available, you can get their info via:
18 error
18 error     npm owner ls typescript-jasmine-istanbul-boilerplate
18 error There is likely additional logging output above.
19 verbose exit [ 1, true ]

```
# Environment
- Windows 10
- node v4.4.7

**Comment by Izhaki** [comment]
Got a strong feeling this has to do with Windows - never tested it on it. I'll have a look at the dependencies. If I remember correctly, one of them had an assertion regarding Windows...

**Comment by Izhaki** [comment]
OK,

First I'm too running this on 4.4.7 on my home machine.

A shot in the dark, but could you try the following line instead of the existing one (I don't have a window environment to test it on):

```
    "test:coverage": "ts-node node_modules/istanbul/lib/cli.js cover -e .ts  -x \"*.d.ts\" -x \"*.spec.ts\" node_modules/jasmine/bin/jasmine.js -- JASMINE_CONFIG_PATH=jasmine.json"
```

The only change is the jasmine path from:

```
node_modules/.bin/jasmine
```

to:

```
node_modules/jasmine/bin/jasmine.js
```

Based on the [Istanbul README.md Usage on Windows section](https://github.com/gotwarlost/istanbul#usage-on-windows).

**Comment by shaunluttin** [comment]
That worked. Well done.

**Comment by Izhaki** [comment]
Repository code not updated yet so re-opening.

**Comment by Izhaki** [comment]
Fixed in ea20ba7c3aa417fd06f8037dee68eb59f6dc151c

**Comment by kayleigh-oliver** [comment]
Thanks. Seems to work with Mocha too.
